plugin/src/main/java/io/jenkins/plugins/redis/RedisFingerprintStorage.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2020, Sumit Sarin and Jenkins project contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package io.jenkins.plugins.redis;
+
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+import hudson.BulkChange;
+import hudson.Extension;
+import hudson.Util;
+import hudson.model.FingerprintStorage;
+import hudson.model.Job;
+import hudson.model.Fingerprint;
+import hudson.util.PersistedList;
+
+import java.io.IOException;
+import java.io.ByteArrayInputStream;
+import java.io.StringWriter;
+import java.io.InputStream;
+import java.util.logging.Logger;
+import java.util.logging.Level;
+
+import jenkins.model.FingerprintFacet;
+import jenkins.util.SystemProperties;
+
+import org.jenkinsci.main.modules.instance_identity.InstanceIdentity;
+
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import redis.clients.jedis.Jedis;
+
+/**
+ * Pluggable external fingerprint storage for fingerprints in Redis.
+ */
+
+@Extension
+public class RedisFingerprintStorage extends FingerprintStorage {
```

Comment:
Does this need to be closable? We want to make sure that Jedis is closed when this class fails.

Reply:
It was indeed a problem, which is why I decided to move to JedisPool, which can offer a thread safe implementation. Would be awesome to hear your thoughts on the new implementation :)